lib/BackbeatConsumer.js

```diff
@@ -546,11 +619,20 @@ class BackbeatConsumer extends EventEmitter {
 
             try {
                 this._consumer.assign(assignment);
+                this._assignment = assignment;
             } catch (e) {
                 // Ignore exceptions if we are not connected
                 const logger = this._consumer.isConnected() ? this._log.error : this._log.debug;
                 logger.bind(this._log)('rdkafka.assign failed', { e: e.toString(), assignment });
+                return;
+            }
+
+            if (this._circuitBreaker.state !== BreakerState.Nominal) {
+                this._pauseAssignments();
             }
+
+            this.emit('assign');
```

Comment:
this is not used?

Reply:
Used by the tests as we need to wait for the partitions to be assigned before sending the circuit breaking event, but i guess we can wait by manually calling `consumer.assignements()`